### PR TITLE
Add a docstring for all in-place Krylov methods

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Review by tmigot — APPROVED** [review]
That's great, thanks!

File: docs/src/solvers/as.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ```@docs
```
Comment (dpo) [inline docs/src/solvers/as.md:1]: Could you add a comment at the top of these doc files to say what they are documenting? I guess "as" stands for "adjoint system"?!

**Review by dpo — COMMENTED** [review]
(my suggestion could happen in another pr)

File: docs/src/solvers/as.md (hunk 1 above)
Comment (amontoison) [reply docs/src/solvers/as.md:4]: I used the trick of https://github.com/JuliaDocs/Documenter.jl/issues/674 to add a comment at the top of these doc files.
I reused the information of `make.jl` : https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/docs/make.jl#L12-L19.

Comment (dpo) [reply docs/src/solvers/as.md:4]: Nice!

**Review by dpo — APPROVED** [review]

**Pull request merged**